hare/hare.go

```diff
@@ -115,19 +125,25 @@ func (h *Hare) isTooLate(id InstanceId) bool {
 var ErrTooLate = errors.New("consensus process %v finished too late")
 
 func (h *Hare) collectOutput(output TerminationOutput) error {
-	id := output.Id()
-
-	if h.isTooLate(id) {
-		return ErrTooLate
-	}
-
 	set := output.Set()
 	blocks := make([]types.BlockID, len(set.values))
 	i := 0
 	for _, v := range set.values {
 		blocks[i] = v.BlockID
 		i++
 	}
+
+	// check validity of the collected output
+	if !h.validate(blocks) {
+		h.Panic("Failed to validate the collected output set")
```

Comment:
should you panic here? why not log and not continue this process?

Reply:
The task is to panic here XD